packages/devtools/bin/devtools.dart

```diff
@@ -48,23 +50,35 @@ void main(List<String> arguments) async {
   }
 
   final bool machineMode = args[argMachine];
+  final port = args[argPort] != null ? int.tryParse(args[argPort]) ?? 0 : 0;
+
   final Uri resourceUri = await Isolate.resolvePackageUri(
       Uri(scheme: 'package', path: 'devtools/devtools.dart'));
   final packageDir = path.dirname(path.dirname(resourceUri.toFilePath()));
+
+  // Default static handler for all non-package requests.
   final String buildDir = path.join(packageDir, 'build');
-  final virDir = new VirtualDirectory(buildDir);
+  final buildHandler =
+      createStaticHandler(buildDir, defaultDocument: 'index.html');
 
-  // Set up a directory handler to serve index.html files.
-  virDir.allowDirectoryListing = true;
-  virDir.directoryHandler = (dir, request) {
-    final indexUri = new Uri.file(dir.path).resolve('index.html');
-    virDir.serveFile(new File(indexUri.toFilePath()), request);
+  // The packages folder is renamed in the pub package so this handler serves
+  // out of the `pack` folder.
+  final String packagesDir = path.join(packageDir, 'build', 'pack');
+  final packHandler =
+      createStaticHandler(packagesDir, defaultDocument: 'index.html');
+
+  // Make a handler that delegates to the correct handler based on path.
+  final handler = (shelf.Request request) {
+    print(request.url.path);
```

Comment:
Is this `print` intentional?

Reply:
Whoops, no - it was me debugging when it didn't work :-D